torch/testing/_internal/common_methods_invocations.py

```diff
@@ -12635,6 +12635,7 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                         DecorateInfo(unittest.expectedFailure,
                                      'TestBinaryUfuncs',
                                      'test_type_promotion'),
+                        DecorateInfo(unittest.expectedFailure, 'TestMeta', 'test_binary_ufuncs_mixed_dtype'),
```

Comment:
I wonder if we could factor `test_type_promotion` into a helper and run it  on meta devices as well? It seems to be a lot more comprehensive.

Or tbh it would be great if we could just run all the opinfo tests with a meta device...

Reply:
at the moment, the tests in `test_meta.py` are written in a very particular way. I agree it would be better to refactor these in more OpInfo friendly.

This is a good onboarding task, I'll keep in mind.